### DISCUSSION TITLE
Automatically Update DNS Rewrites for Local DHCP Hostnames

### DISCUSSION THREAD

**Discussion by Junto026** [AdguardTeam/AdGuardHome#6040] [Q&A] [answered]
I use AdGuardHome as my DHCP and DNS server, and I have a domain registered `mydomain.com`.  I want to be able to associate each device in my network an IP address, it's hostname (e.g. `printer.mydomain.com`) and then have any device in my network be able to resolve it.

I can sort of accomplish it by creating a static DHCP record for the printer, and give it the hostname `printer.mydomain.com`.  But the only way it resolves is if another device requests `printer.mydomain.com.lan`.  I know I could add a second DNS re-write entry that points `printer.mydomain.com` back to the IP address of the printer, but that means I'd have to have 2 things configured for every device (it's DHCP entry _and_ its DNS re-write).  

Is there a way to do this without having two configurations for every device?

**Comment by ppfeufer** [comment, marked as answer]
If I'm not completely mistaken, that can be configured in your `AdGuardHome.yaml` file in the `dhcp` secrtion.

```yaml
 dhcp:
   local_domain_name: mydomain.com
 ```

**Reply by Junto026** [reply]
That did it, thank you!! My last step is to figure out how to add a DHCP option to send the search domain to all the clients. I suspect I can figure that out through existing documentation. Thanks so much for the quick reply and I’m glad it was so simple.
